### PR TITLE
Added a really basic form component

### PR DESCRIPTION
**Pull request by crain9412**

### CODE REVIEW

File: src/components/Form/Form.stories.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { storiesOf, action } from '@storybook/react';
+import Form from './index';
+import * as FormConstants from './constants';
```
Comment (kevinmichaelchen) [inline src/components/Form/Form.stories.js]: I'd avoid wildcard imports where possible, especially for an API that others will use. Also, functions are nicer than constants. Here's the kind of API I'm envisioning:
```es6
// this is kind of like how we do 
// import { call } from 'redux-saga/effects'; 
import { createTextField, createTextAreaField, createCheckboxField } from '@teslagov/react-components/form';

const formFields = [
  createTextField("firstName", "First Name"),
  createTextField("lastName", "Last Name"),
  createTextAreaField("justification", "Justification"),
  createCheckboxField("anonymous", "Anonymous")
];

formFields.forEach(e => e.onChange = action(`Changed ${e.label}`))

storiesOf('Form', module)
  .add('Example', () => 
    <Form title={title} 
          fields={formFields}
          onSubmit={action('submittedForm')}/>
  );
```

File: src/components/Form/Form.stories.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+form.forEach( (element) => {
+  element.onChange = action('Changed ' + element.label );
+} );
+
+storiesOf('Form', module)
+  .add('Example', () =>
+    <Form
+      title={title}
+      form={form}
```
Comment (kevinmichaelchen) [inline src/components/Form/Form.stories.js]: I'd rename `form` prop to `fields`, like in the above example.

**Comment by crain9412** [comment]
We use redux-form on our other projects, do you think it would be better to add a redux-form dependency to this project and use their `Field` instead of creating our own components for each form element? @kevinmichaelchen

**Comment by kevinmichaelchen** [comment]
@crain9412 @MarcusTHellams @fitzyjoe @bilal1323 In [clarakm-login issue #1](https://github.com/TeslaGov/clarakm-login/pull/1/files) I incorporated the [clarakm-form](https://github.com/TeslaGov/clarakm-form] library. We can now make form fields with the `createField` function the library exposes:
```es6
import {
  createField,
  required,
  maxLength
} from '@teslagov/clarakm-form';

<form onSubmit={handleSubmit}>
      {
        createField({
          type: 'email',
          name: 'emailAddress',
          validate: [required, maxLength(255)],
          label: 'Email',
          labelId: 'email-address-label',
          labelClassName: 'mb-0',
          inputId: 'login-input',
          inputClassName: 'form-control login-input py-1',
          autoFocus: true
        })
      }
      {
        createField({
          type: 'password',
          name: 'password',
          label: 'Password',
          labelId: 'password-label',
          labelClassName: 'mb-0',
          inputId: 'password-input',
          inputClassName: 'form-control login-input py-1'
        })
      }
    </form>
```

**Pull request closed without merging**